File: src/main/java/com/dongtronic/diabot/data/nutrition/Tags.java

package com.dongtronic.diabot.data.nutrition;

import java.util.HashMap;
import java.util.Map;
import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "item",
    "measure",
    "quantity",
    "food_group",
    "tag_id"
})
public class Tags {

    @JsonProperty("item")
    private String item;
    @JsonProperty("measure")
    private String measure;
    @JsonProperty("quantity")
    private String quantity;
    @JsonProperty("food_group")
    private Integer foodGroup;
    @JsonProperty("tag_id")
    private Integer tagId;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("item")
    public String getItem() {
        return item;
    }

    @JsonProperty("item")
    public void setItem(String item) {
        this.item = item;
    }

    @JsonProperty("measure")
    public String getMeasure() {
        return measure;
    }

    @JsonProperty("measure")
    public void setMeasure(String measure) {
        this.measure = measure;
    }

    @JsonProperty("quantity")
    public String getQuantity() {
        return quantity;
    }

    @JsonProperty("quantity")
    public void setQuantity(String quantity) {
        this.quantity = quantity;
    }

    @JsonProperty("food_group")
    public Integer getFoodGroup() {
        return foodGroup;
    }

    @JsonProperty("food_group")
    public void setFoodGroup(Integer foodGroup) {
        this.foodGroup = foodGroup;
    }

    @JsonProperty("tag_id")
    public Integer getTagId() {
        return tagId;
    }

    @JsonProperty("tag_id")
    public void setTagId(Integer tagId) {
        this.tagId = tagId;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: bot/src/main/java/com/dongtronic/diabot/data/nutrition/Tags.kt
package com.dongtronic.diabot.data.nutrition

import com.fasterxml.jackson.annotation.*
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("item", "measure", "quantity", "food_group", "tag_id")
class Tags {
    @get:JsonProperty("item")
    @set:JsonProperty("item")
    @JsonProperty("item")
    var item: String? = null
    @get:JsonProperty("measure")
    @set:JsonProperty("measure")
    @JsonProperty("measure")
    var measure: String? = null
    @get:JsonProperty("quantity")
    @set:JsonProperty("quantity")
    @JsonProperty("quantity")
    var quantity: String? = null
    @get:JsonProperty("food_group")
    @set:JsonProperty("food_group")
    @JsonProperty("food_group")
    var foodGroup: Int? = null
    @get:JsonProperty("tag_id")
    @set:JsonProperty("tag_id")
    @JsonProperty("tag_id")
    var tagId: Int? = null
    @JsonIgnore
    private val additionalProperties: MutableMap<String, Any> = HashMap()

    @JsonAnyGetter
    fun getAdditionalProperties(): Map<String, Any> {
        return additionalProperties
    }

    @JsonAnySetter
    fun setAdditionalProperty(name: String, value: Any) {
        additionalProperties[name] = value
    }
}
